sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/AbstractCompilerCommand.java

```diff
@@ -3,121 +3,103 @@
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
-import java.util.Set;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.lang3.tuple.Pair;
 import picocli.CommandLine;
 import picocli.CommandLine.ScopeType;
 
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.Arrays;
-import java.util.function.Predicate;
-import java.util.function.Supplier;
-
-import static com.datasqrl.packager.config.ScriptConfiguration.GRAPHQL_NORMALIZED_FILE_NAME;
-
 @Slf4j
 public abstract class AbstractCompilerCommand extends AbstractCommand {
 
   public static final Path DEFAULT_DEPLOY_DIR = Path.of("build", "deploy");
-  protected final boolean execute;
-  protected final boolean startGraphql;
 
   @CommandLine.Parameters(arity = "1..2", description = "Main script and (optional) API specification")
   protected Path[] files;
 
-  @CommandLine.Option(names = {"-a", "--api"}, description = "Generates the API specification for the given type")
+  @CommandLine.Option(names = {"-a", "--api"},
+      description = "Generates the API specification for the given type")
   protected APIType[] generateAPI = new APIType[0];
 
-  @CommandLine.Option(names = {"-d", "--debug"}, description = "Outputs table changestream to configured sink for debugging")
+  @CommandLine.Option(names = {"-d", "--debug"},
+      description = "Outputs table changestream to configured sink for debugging")
   protected boolean debug = false;
 
-  @CommandLine.Option(names = {"-t", "--target"}, description = "Target directory for deployment artifacts")
+  @CommandLine.Option(names = {"-t", "--target"},
+      description = "Target directory for deployment artifacts")
   protected Path targetDir = DEFAULT_DEPLOY_DIR;
 
   @CommandLine.Option(names = {"--mnt"}, description = "Local directory to mount for data access")
   protected Path mountDirectory = null;
 
-  @CommandLine.Option(names = {"--nolookup"}, description = "Do not look up package dependencies in the repository",
+  @CommandLine.Option(names = {"--nolookup"},
+      description = "Do not look up package dependencies in the repository",
       scope = ScopeType.INHERIT)
   protected boolean noinfer = false;
-  private boolean kafkaStarted;
 
-  protected AbstractCompilerCommand(boolean execute, boolean startGraphql, boolean startKafka) {
-    this.execute = execute;
-    this.startGraphql = startGraphql;
-    this.startKafka = startKafka;
-  }
+  @CommandLine.Option(names = {"-p", "--profile"},
+      description = "An alternative set of configuration values which override the default package.json")
+  protected String[] profiles = new String[0];
+
+  @SneakyThrows
+  public void execute(ErrorCollector errors) {
+    Repository repository = createRepository(errors);
 
-  public void runCommand(ErrorCollector errors) {
-    setupTargetDir();
+    PackageBootstrap packageBootstrap = new PackageBootstrap(root.rootDir,
```

Comment:
Nice, I like the breakdown into bootstrap -> config -> packager -> build directory.